beacon/src/beacon.rs

```diff
@@ -20,34 +21,53 @@ pub struct Beacon {
 }
 
 impl Beacon {
+    /// Construct a new beacon with a given remote and local entropy. The remote
+    /// and local entropy are checked for version equality.
+    ///
+    /// Version 0 beacons use a Sha256 of the remote and local entropy (data and
+    /// timestamp), which is then used as a 32 byte seed to a ChaCha12 rng. This
+    /// rng is used to choose a random frequency from the given region
+    /// parameters and a payload size between MIN_BEACON_V0_PAYLOAD_SIZE and
+    /// MAX_BEACON_V0_PAYLOAD_SIZE.
     pub fn new(
         remote_entropy: Entropy,
         local_entropy: Entropy,
         region_params: &[BlockchainRegionParamV1],
     ) -> Result<Self> {
-        let mut data = {
-            let mut hasher = Sha256::new();
-            remote_entropy.digest(&mut hasher);
-            local_entropy.digest(&mut hasher);
-            hasher.finalize().to_vec()
-        };
-        let mut seed = [0u8; 32];
-        seed.copy_from_slice(&data[0..32]);
-        let mut rng = rand_chacha::ChaCha12Rng::from_seed(seed);
+        if remote_entropy.version != local_entropy.version {
+            return Err(Error::invalid_version());
```

Comment:
I'm not sure about this check, what happens if the code support entropy 0 and 1, and we switch it on the server, the client cannot allow both versions to exist across the update?

Reply:
I'll just drop local version then.. It's really not relevant what the version is